worker/taskstatusupdate.go

```diff
@@ -45,42 +47,51 @@ func reportFailed(client client.Queue, task *TaskRun, log *logrus.Entry) *update
 }
 
 func reportCompleted(client client.Queue, task *TaskRun, log *logrus.Entry) *updateError {
-	_, _, err := client.ReportCompleted(task.TaskID, strconv.FormatInt(int64(task.RunID), 10))
+	_, err := client.ReportCompleted(task.TaskID, strconv.FormatInt(int64(task.RunID), 10))
 	if err != nil {
 		log.WithField("error", err).Warn("Not able to report successful completion for task.")
 		return &updateError{err: err.Error()}
 	}
 	return nil
 }
 
-func claimTask(client client.Queue, taskID string, runID int, workerID string, workerGroup string, log *logrus.Entry) (*taskClaim, *updateError) {
+func claimTask(client client.Queue, taskID string, runID int, workerID string, workerGroup string, log *logrus.Entry) (*taskClaim, error) {
 	log.Info("Claiming task")
 	payload := queue.TaskClaimRequest{
 		WorkerGroup: workerGroup,
 		WorkerID:    workerID,
 	}
 
-	tcrsp, callSummary, err := client.ClaimTask(taskID, strconv.FormatInt(int64(runID), 10), &payload)
+	tcrsp, err := client.ClaimTask(taskID, strconv.FormatInt(int64(runID), 10), &payload)
 	// check if an error occurred...
 	if err != nil {
-		e := &updateError{
-			err:        err.Error(),
-			statusCode: callSummary.HttpResponse.StatusCode,
-		}
-		var errorMessage string
-		switch {
-		case callSummary.HttpResponse.StatusCode == 401 || callSummary.HttpResponse.StatusCode == 403:
-			errorMessage = "Not authorized to claim task."
-		case callSummary.HttpResponse.StatusCode >= 500:
-			errorMessage = "Server error when attempting to claim task."
+		switch err := err.(type) {
```

Comment:
Ah, so we did need it one place :)

Probably later when we adopt formalized errors we can look into changing this too.....